### ISSUE TITLE
File Type is now empty when updating (in 3.3)

### ISSUE THREAD

**Issue by skornev** [symfony/symfony#24792]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | yes
| RFC?             | no
| Symfony version  | 3.3.10

After upgrading from 3.2 to 3.3.  Files collection is now just empty in controller when updating entity without selecting new files. So it just removes the file entities. It used to work before 3.3 now it fails. 
Maybe I missed something and it's  planned changes in form file type behavior?

Here is a demo which shows the issue  https://github.com/skornev/vichbundle-sandbox-not-cp-symfony3.3

This issue was raised here https://github.com/dustin10/VichUploaderBundle/issues/776.

**Comment by xabbuh** [comment]
could be related to the changes in #24198 and #24606

ping @enumag

**Comment by yceruto** [comment]
https://github.com/symfony/symfony/pull/24606 has not been released yet, so I guess it'll be fixed for 3.3.11

**Comment by enumag** [comment]
I believe this is a duplicate of #24546 and is fixed by #24606.

@skornev Can you try the change I made in #24606 to confirm?

**Comment by skornev** [comment]
Yes, the changes in FileBag  work but there comes another issue. Saving and updating work like a charm. 
But If we add another file input without choosing any file,  it fills collection with an entity with null values.
(and later doctrine raises error because my schema does not allow null value)
Should it be treated like a normal behavior and we should check this collection and manually remove those entities with null values or it's a bug?

**Comment by enumag** [comment]
I'm not entirely sure but it's possibly intended. If you update your demo to showcase this new problem I could check it.

**Comment by skornev** [comment]
It's the same demo but with your patch applied to the FileBag class

**Comment by enumag** [comment]
After #24606 the behaviour is the same as it was before #24198 (meaning symfony 3.3.9) so it's not something that was changed by my commit in 3.3.10. I believe it's indeed intentional.

**Comment by skornev** [comment]
Yes, @enumag you were right . As for the last issue, I have just checked on 3.2 and behavior is the same.  Looks like it's intentional behavior.

**Comment by xabbuh** [comment]
closing here as fixed by #24606 then
